src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ApplicationTime;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    /**
+     * @return array[]
+     */
+    public function findActivePromotionsByPid(Pid $pid, ?int $limit, int $offset): array
```

Comment:
I *think* when we get promotionOfCoreEntity, we're also going to want to populate all of that programme's ancestry too. See if you can find an example of where a promotion of a programme, and see how it renders on the frontend - do we need it's parent's title information?